tests/osquery/test_osquery_api.py

```diff
@@ -1405,3 +1407,18 @@ def test_public_urls(self):
                 reverse(f"osquery_public:{route}"),
                 url_prefix + reverse(f"osquery_public_legacy:{route}")
             )
+
+    def test_mount_legacy_public_endpoints_flag(self):
+        url_prefix = "/public"
+        routes = ['enroll', 'config', 'carver_start', 'carver_continue', 'distributed_read', 'distributed_write']
+
+        urlpatterns_w_legacy = tuple(build_urlpatterns_for_zentral_apps(mount_legacy_public_endpoints=True))
+        urlpatterns_wo_legacy = tuple(build_urlpatterns_for_zentral_apps(mount_legacy_public_endpoints=False))
```

Comment:
To be able to remove the argument, and still test, you can dynamically modify the zentral_config.

Reply:
Check how did I do it please.